apis/ingress/v1/pomerium_types.go

```diff
@@ -283,6 +283,12 @@ type PomeriumSpec struct {
 
 	// UseProxyProtocol enables <a href="https://www.pomerium.com/docs/reference/use-proxy-protocol">Proxy Protocol</a> support.
 	UseProxyProtocol *bool `json:"useProxyProtocol,omitempty"`
+
+	// CodecType sets the <a href="https://www.pomerium.com/docs/reference/codec-type">Codec Type</a>.
+	//
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Enum=auto;http1;http2;http3
+	CodecType *string `json:"codec_type,omitempty"`
```

Comment:
Should the default be `auto` or is there a difference between unset and `auto`?

Reply:
currently unset will end up as auto

Reply:
it might be better to have the options be "http1" "http2" or "http3" and unset is equivalent to auto.

Reply:
This is a change in requirements. It means updating docs, core, console and zero. Are you sure you want me to do this?

Reply:
Oh, nah it's not that important. It's fine as is